### PR TITLE
Refresh prices when app becomes active

### PR DESCRIPTION
**Pull request by michelgoni**
## Summary
- reload pricing snapshot when the app becomes active
- keep notification authorization refresh behavior intact
- cover foreground refresh with AppFeature test dependencies

## Tests
- xcodebuild -project PrecioLuzApp.xcodeproj -scheme PrecioLuzApp -destination 'platform=iOS Simulator,name=iPhone 17 Pro Max' test -only-testing:PrecioLuzAppTests/AppFeatureTests
- xcodebuild -project PrecioLuzApp.xcodeproj -scheme PrecioLuzApp -destination 'platform=iOS Simulator,name=iPhone 17 Pro Max' test -only-testing:PrecioLuzAppTests/REEPublicationWindowPolicyTests -only-testing:PrecioLuzAppTests/DailyPricingSnapshotPipelineTests
- pre-push hook: build + 115 tests passed

### CODE REVIEW

File: Sources/Features/PricesFeature/PricesView.swift [hunk 1]
```diff
@@ -85,14 +87,61 @@ struct PricesView: View {
...
+    ) -> some View {
+        if !state.isLoading {
+            PricesMediumWidgetView(model: mediumWidgetModel(now: now, calendar: calendar))
+                .frame(maxWidth: .infinity, alignment: .center)
+                .padding(.top, PricesViewLayout.widgetTopPadding)
+                .accessibilityElement(children: .contain)
+                .accessibilityIdentifier("pricesMediumWidgetContainer")
+                .transition(.opacity.combined(with: .move(edge: .top)))
+
+            summaryContent
```
[the remark this answers is not in the document]
Comment (michelgoni) [reply Sources/Features/PricesFeature/PricesView.swift]: Fixed in b1bb440. The summary cards now receive a timeline-derived summary, recomputing the `current` price from the active TimelineView date while preserving the loaded daily average/min/max metrics. Added a regression test for the hour-boundary case.
